### PR TITLE
[ICP-8986] Provide change for EcoNet Bulldog

### PR DESCRIPTION
**Pull request by MMateusiakS**
@dkirker  and @kianooshST, 
Bulldog valve had issue that during opening randomly was sending BinaryReport with wrong value. For this valve only BasicGet should be handled.

### CODE REVIEW

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy [hunk 1]
```diff
@@ -102,32 +102,42 @@ def zwaveEvent(physicalgraph.zwave.commands.deviceresetlocallyv1.DeviceResetLoca
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
-	def value = cmd.value == 0xFF ? "open" : cmd.value == 0x00 ? "closed" : "unknown"
 
-	return [createEventWithDebug([name: "contact", value: value, descriptionText: "$device.displayName valve is $value"]),
-			createEventWithDebug([name: "valve", value: value, descriptionText: "$device.displayName valve is $value"])]
+	if(!isBulldogValve()) {
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: Please include a space after the `if`

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy [hunk 2]
```diff
@@ -102,32 +102,42 @@ def zwaveEvent(physicalgraph.zwave.commands.deviceresetlocallyv1.DeviceResetLoca
...
 
 def open() {
-	delayBetween([
-		zwave.basicV1.basicSet(value: 0xFF).format(),
-		zwave.switchBinaryV1.switchBinaryGet().format()
-	], 10000) //wait for a water valve to be completely opened
+	def cmd = []
+	cmd << zwave.basicV1.basicSet(value: 0xFF).format()
+	
+	if(!isBulldogValve()) {
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: Please include a space after the `if`

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy [hunk 3]
```diff
@@ -102,32 +102,42 @@ def zwaveEvent(physicalgraph.zwave.commands.deviceresetlocallyv1.DeviceResetLoca
...
-		zwave.switchBinaryV1.switchBinaryGet().format()
-	], 10000) //wait for a water valve to be completely closed
-}
-
-def poll() {
-	zwave.switchBinaryV1.switchBinaryGet().format()
+	def cmd = []
+	cmd << zwave.basicV1.basicSet(value: 0x00).format()
+	
+	if(!isBulldogValve()) {
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: Please include a space after the `if`

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy [hunk 4]
```diff
@@ -139,7 +149,14 @@ def ping() {
 
 def refresh() {
 	log.debug "refresh() is called"
-	def commands = [zwave.switchBinaryV1.switchBinaryGet().format()]
+	def commands = []
+	
+	if(!isBulldogValve()) {
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: Please include a space after the `if`

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy [hunk 5]
```diff
@@ -80,6 +79,7 @@ def parse(String description) {
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: Not sure that an empty line here is necessary. Let's go ahead and remove it.

**Review by dkirker — CHANGES_REQUESTED** [review]
Other than a few code formatting things, this looks good. Clean those up and as long as it tests good, I see no issue. @kianooshST Any objections?

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy (hunk 1 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: ok

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy (hunk 2 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: ok

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy (hunk 3 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: ok

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy (hunk 4 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: ok

File: devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy (hunk 5 above)
Comment (MMateusiakS) [reply devicetypes/smartthings/zwave-water-valve.src/zwave-water-valve.groovy]: ok

**Comment by MMateusiakS** [comment]
@dkirker fixes added

**Comment by kianooshkarami** [comment]
Can you please describe the problem with more description? 

I am ok with the change, has the manufacturer been contacted? Do you have logs that describe the problem with more details (hub logs preferably)?

**Comment by MMateusiakS** [comment]
We walked around issue in DTH, not to contact Company. If you want we can collect logs.

**Comment by kianooshkarami** [comment]
Let's please capture logs, and describe the problem so we can contact the manufacturer. 

I do not want to open up a can of worms here, but I really would like to document why these manufacturer specific changes are added for sake of the future changes.

**Pull request closed without merging**